File: Backend/src/ShoeStore.Application/DTOs/VoucherDtos/CreateVoucherDto.cs

namespace ShoeStore.Application.DTOs.VoucherDtos
{
    public class CreateVoucherDto
    {
        public string? VoucherName { get; set; }
        public string? VoucherDescription { get; set; }
        public decimal? Discount { get; set; }
        public int VoucherScope { get; set; } = 1; // Default to AllProducts
        public int DiscountType { get; set; } = 1; // Default to Percentage 
        public decimal MaxPriceDiscount { get; set; }
        private DateTime? _validFrom;
        public DateTime? ValidFrom
        {
            get => _validFrom;
            set => _validFrom = value.HasValue ? DateTime.SpecifyKind(value.Value, DateTimeKind.Utc) : null;
        }

        private DateTime? _validTo;
        public DateTime? ValidTo
        {
            get => _validTo;
            set => _validTo = value.HasValue ? DateTime.SpecifyKind(value.Value, DateTimeKind.Utc) : null;
        }
        public int? MaxUsagePerUser { get; set; }
        public int? TotalQuantity { get; set; }
        public decimal? MinOrderPrice { get; set; }
    }
}

File: Frontend/app/src/main/java/com/example/shoestoreapp/features/admin/voucher/data/remote/CreateVoucherDto.kt
package com.example.shoestoreapp.features.admin.voucher.data.remote

import com.google.gson.annotations.SerializedName

data class CreateVoucherDto(
    @SerializedName("voucherName") val voucherName: String,
    @SerializedName("voucherDescription") val voucherDescription: String,
    @SerializedName("voucherScope") val voucherScope: Int,
    @SerializedName("discountType") val discountType: Int,
    @SerializedName("discount") val discount: Double,
    @SerializedName("maxPriceDiscount") val maxPriceDiscount: Double,
    @SerializedName("minOrderPrice") val minOrderPrice: Double,
    @SerializedName("totalQuantity") val totalQuantity: Int,
    @SerializedName("maxUsagePerUser") val maxUsagePerUser: Int,
    @SerializedName("releaseType") val releaseType: Int,
    @SerializedName("validFrom") val validFrom: String,
    @SerializedName("validTo") val validTo: String
)
